src/core/execute.c

```diff
@@ -1205,6 +1205,10 @@ static int setup_pam(
         if (pam_code != PAM_SUCCESS)
                 goto fail;
 
+        pam_code = pam_setcred(handle, PAM_ESTABLISH_CRED | PAM_SILENT);
+        if (pam_code != PAM_SUCCESS)
+                goto fail;
```

Comment:
uh, so first of all the "flags" field already contains PAM_SILENT if appropriate and should be used here. 

But the bigger problem is this: we invoke setgroups() later which will then override any groups set here again, replacing everything. I figure if we allow pam_setcreds(), then this should allow additive group memberships that do not conflict with our SupplementaryGroups= setting.

Hence, I figure the code needs to be more complex here: there needs to be some clearly defined logic what happens in the four cases:

1. PAM and SupplementaryGroups= is used together
2. only PAM is used
3. only SupplementaryGroups= is used
4. neither is used.

Currently case 3 and 4 are clearly defined: in case 3 *only* the supplementary groups will be in effect after, and in case 4 group memberships are not touched, and inherited from PID 1, which generally means there are none. I have the suspicion it would make most sense if we'd explicitly clear the groups list before invoking pam_setcreds(), then query the resulting list and merge it with the list of SupplementaryGroups=. But then there also needs to be a strategy if we run inside a container that takes the ability to drop group memberships away (i.e. the stuff #4280 added)...